weatherApp/src/screens/AddCityScreen/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import {Text, SafeAreaView, View, TouchableOpacity} from 'react-native';
+import SearchBar from '../../components/Searchbar';
+import styles from './styles';
+import {NativeStackScreenProps} from '@react-navigation/native-stack';
+import {RootStackParamList} from '../CitiesStackScreen';
+import RenderIconButton from '../../components/RenderIconButton';
+import {FlatList} from 'react-native-gesture-handler';
+import {CityResponse, citiesApi} from '../../services/CitiesAPI';
+import {City} from '../../stores/CityStore';
+import DraggableFlatList from 'react-native-draggable-flatlist';
+import {toJS} from 'mobx';
+import {Stack} from 'react-native-spacing-system';
+import {useCityStore, useLanguageStore} from '../../contexts/StoreContext';
+import {observer} from 'mobx-react';
+import RenderFavouriteCitiesItem from '../../components/RenderFavouriteCitiesItem';
+
+type AddCityScreenProps = NativeStackScreenProps<
+  RootStackParamList,
+  'AddCityScreen'
+>;
+
+const AddCityScreen = (props: AddCityScreenProps) => {
+  const cityStore = useCityStore();
+  const languageStore = useLanguageStore();
+
+  const [searchbarCityValue, setSearchbarCityValue] = useState('');
+  const [isSearchBarFocused, setIsSearchBarFocused] = useState(false);
+  const [foundCities, setFoundCities] = useState<CityResponse[]>([]);
+
+  const handleClose = useCallback(() => {
+    props.navigation.goBack();
+  }, [props.navigation]);
+
+  useEffect(() => {
+    props.navigation.setOptions({
+      headerRight: () =>
+        RenderIconButton({
+          name: 'close',
+          size: 30,
+          color: 'black',
+          onPress: handleClose,
+        }),
+    });
+  }, [props.navigation, handleClose]);
+
+  useEffect(() => {
+    if (searchbarCityValue) {
```

Comment:
It's a pretty expensive operation. Usually, it's done in the following way: a user inputs value, and if he stops to input (for example, during 1 second there are no more inputs), the request is sent.